lib/ansible/modules/system/known_hosts.py

```diff
@@ -240,7 +243,7 @@ def search_for_host_key(module,host,key,hash_host,path,sshkeygen):
 
     sshkeygen_command.insert(1,'-H')
     rc,stdout,stderr=module.run_command(sshkeygen_command,check_rc=False)
-    if rc!=0: #something went wrong
+    if rc not in (0, 1) or stderr != '': #something went wrong
```

Comment:
BTW https://github.com/ansible/ansible/commit/dd9d5cc670eccf3c92c8bf974dd294787fe94169 was a workaround for a very similar issue.  And then https://github.com/ansible/ansible/commit/405c6820b0afd74fd335d2d302028e7e816f5bb9 added a 2nd invocation of ssh-keygen, but was apparently never tested with newer OpenSSH versions.